doc/releases/changelog-dev.md

```diff
@@ -603,6 +603,9 @@
 
 <h3>Bug fixes 🐛</h3>
 
+* Remove a redundant `functools.lru_cache` call that was capturing all `SampleMP` objects in a workload.
+  [(#7077)](https://github.com/PennyLaneAI/pennylane/pull/7077)
```

Comment:
@mlxd could you add a first sentence here to the bugfix, to indicate to a reader what has improved/fixed from their perspective?

Could be as simple as 'dynamic `one-shot` MCM executions are now faster', or something more general if you know the impact top-of-head.

This will help us better market the changes this release to users.

Reply:
It's fine to keep what you have as the second sentence/technical explanation of the change